onnxruntime/core/graph/model.cc

```diff
@@ -284,6 +284,38 @@ Status Model::Load(std::unique_ptr<ModelProto> p_model_proto, std::shared_ptr<Mo
   return Status::OK();
 }
 
+template <typename T>
+static Status LoadModel(const T& file_path, ONNX_NAMESPACE::ModelProto& model_proto) {
+  int fd;
+  Status status = Env::Default().FileOpenRd(file_path, fd);
+  if (!status.IsOK()) {
+    if (status.Category() == common::SYSTEM) {
+      switch (status.Code()) {
+        case ENOENT:
+          return ORT_MAKE_STATUS(ONNXRUNTIME, NO_SUCHFILE, "Load model ", ToMBString(file_path),
+                                 " failed. File doesn't exist");
+        case EINVAL:
+          return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Load model ", ToMBString(file_path), " failed");
+        default:
+          return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "system error number ", status.Code());
+      }
+    }
+  }
+  try {
+    status = Model::Load(fd, model_proto);
+  } catch (std::exception& ex) {
+    GSL_SUPPRESS(es .84)
+    ORT_IGNORE_RETURN_VALUE(Env::Default().FileClose(fd));
```

Comment:
>ORT_IGNORE_RETURN_VALUE(Env::Default().FileClose(fd)); [](start = 4, length = 54)

There is not a finally clause in C++. You need to your RAII.

Reply:
Same comment as above.

Reply:
This is not a new change as will be visible by the latest commit. I removed the code duplication which implicitly implied that I added this code. We can fix them in a separate PR.